metrics/cli/docs/features/doctor_output_improvements/diagrams/ci_integrations_validator_class_diagram.puml

```diff
@@ -0,0 +1,114 @@
+@startuml ci_integrations_validator_class_diagram
+
+package core.lib.src.features.validation {
+    package model {
+        package builder {
+            class ValidationResultBuilder {
+                - _results: UnmodifiableMapView<ValidationTarget, TargetValidationResult>
+
+                + setResult(result: TargetValidationResult): void
+                + setEmptyResults(result: TargetValidationResult): void
+                + build(): ValidationResult
+            }
+        }
+
+        class ValidationTarget<T> {
+            + data: T
```

Comment:
🔨
Do we actually need the `data` field in this class? Maybe we can remove it from here?